hetmech/matrix.py

```diff
@@ -69,5 +76,8 @@ def normalize(matrix, vector, axis, damping_exponent):
         vector **= -damping_exponent
     vector[numpy.isinf(vector)] = 0
     shape = (len(vector), 1) if axis == 'rows' else (1, len(vector))
-    matrix *= vector.reshape(shape)
+    if sparse.issparse(matrix):
+        matrix = matrix.multiply(vector.reshape(shape))
```

Comment:
`vector.reshape(shape)` repeated... just do the reshaped to vector outside of if statement?